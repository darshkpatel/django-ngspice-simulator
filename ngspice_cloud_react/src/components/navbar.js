import React from "react";
import "./css/navbar.css"
import {
  Navbar,
  NavbarToggler,
  NavbarBrand,
  Nav,
  Collapse
} from "shards-react";

class FullNavBar extends React.Component {
  constructor(props) {
    super(props);

    this.toggleNavbar = this.toggleNavbar.bind(this);

    this.state = {
      collapseOpen: false
    };
  }


  toggleNavbar() {
    this.setState({
      ...this.state,
      ...{
        collapseOpen: !this.state.collapseOpen
      }
    });
  }

  render() {
    return (
      <Navbar type="dark" theme="primary" expand="md">
        <NavbarBrand href="#">ngSpice Cloud</NavbarBrand>
        <NavbarToggler onClick={this.toggleNavbar} />

        <Collapse open={this.state.collapseOpen} navbar>
          <Nav navbar>



          </Nav>


        </Collapse>
      </Navbar>
    );
  }
}

export default FullNavBar;