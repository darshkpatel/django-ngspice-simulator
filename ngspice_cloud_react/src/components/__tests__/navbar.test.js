
// Auto-generated do not edit


/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import navbar from '../navbar';


describe('navbar test', () => {
  it('navbar should match snapshot', () => {
    const component = renderer.create(<navbar
       />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
