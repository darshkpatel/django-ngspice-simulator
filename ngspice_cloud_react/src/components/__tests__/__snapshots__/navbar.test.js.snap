// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navbar test navbar should match snapshot 1`] = `<navbar />`;
